recipe/conda_forge_ci_setup/upload_or_check_non_existence.py

```diff
@@ -168,18 +160,20 @@ def upload_or_check(recipe_dir, owner, channel, variant):
     # This is the actual fix where we create the token file once and reuse it
     # for all uploads
     if token:
-        (
-            _should_validate,
-            request_copy
-        ) = _import_feedstock_outputs_functions(recipe_dir)
-
         with get_temp_token(cli.token) as token_fn:
             for path in new_distributions:
                 upload(token_fn, path, owner, channel)
                 print('Uploaded {}'.format(path))
 
             if _should_validate():
-                return request_copy([b[2] for b in built_distributions], channel)
+                for path in existing_distributions:
+                    delete_dist(token_fn, path, owner, channel)
```

Comment:
What's the use of this? I don't understand

Reply:
well I could be wrong but I thought anaconda uploads fail if the package already exists. 

For the staging org, we always want to upload the build we have even if it is already on the org. We have to do this so that the checksums we send match the files we then request a copy dornn

Reply:
This is problematic. There are some package with a non-noarch output and a noarch output. Noarch output is built on all platform and is basically the same for all platforms but they are not equal bit-to-bit. Whichever platform uploads first wins. Depending on who uploads first and who requests copy first, this might fail.

Reply:
I’m not following this. Can you point me to an example?

Reply:
Just to be clear, all of the stuff here happens with a platform. Also this code doesn’t remove all files for a given package, just one of them.

Reply:
I deleted my other comments since upon further thought I am following this. 

IMHO this is a bug in the package and not the code here. A given recipe should produce an output only once and upload it only once so we always know how things were built. 

In terms of a fix now, I’m not sure how to proceed. If we can make a list of the packages to fix, then we can keep the code here and fix the packages. 

Otherwise, idk how to allow for this in our current framework. Any system where the package and the thing to verify it come in two parts or are not isolated in the same transaction has this race condition.

Reply:
We could not upload noarch packages on non-linux-64 platforms. As long as every noarch package is built in at least Linux-64, then this should work.

Reply:
We could have smithy pick which  noarch package to upload and record this when rendering by having it set an env var. then we can use the env var to test if we should handle noarch packages in this code.